.github/workflows/check-projects.yml

```diff
@@ -0,0 +1,329 @@
+#
+# Analysis goals
+#
+# The check-projects workflow builds a Spotbugs runnable from master (labelled baseline) and the current PR (labelled new).
+# The unit tests are *not* run in this job on purpose to keep it simple and allow for a fast response time.
+# After building the baseline and new versions of Spotbugs, an analysis is run on open source projects.
+# The list of projects currently analyzed:
+#   - spotbugs
+#   - matsim-libs
+#   - jenkins
+#
+# The workflow analyses these projects via the baseline and the new versions of Spotbugs and produces a diff of analysis results for each one.
+# The diff of the analysis results can be used to see if the PR produces new results and/or old results are gone.
+# Time of analysis is also measured in order to catch performance regressions.
+#
+# Installation
+#
+# These workflows are intended to be installed on a Spotbugs fork.
+# On the default branch of the repository (main or master by default) the workflow description files should be placed into the `.github/workflows` directory.
+# Both check-projects.yml and check-projects-comment files should be there and must be committed.
+#
+# Usage
+#
+# Once installed you can use the workflows by creating a feature branch in your fork, and creating a pull-request from that branch to your fork's master branch.
+# The outcome of the analysis is seen on the job status of the PR, and the diff of the results can be seen as PR comment.
+
+name: check-projects
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check-spotbugs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Checkout Baseline
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.base.ref }}
+          path: spotbugs-baseline
+          persist-credentials: false
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v1
+        with:
+          java-version: '17.0.x'
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Quick-build new Spotbugs
+        run: ./gradlew installDist --no-daemon
+
+      - name: Quick-build baseline Spotbugs
+        run: ./gradlew installDist --no-daemon
+        working-directory: spotbugs-baseline
+
+      - name: Cache analyzed Spotbugs
+        uses: actions/cache@v2
+        with:
+          path: spotbugs-4.4.1.jar
```

Comment:
Could you move the version infos to variables, so it's easier to modify them in the future? (Not only for spotbugs, but the other projects as well.)